pkg/controller/template/template_controller.go

```diff
@@ -487,12 +487,12 @@ func createNewCert(cert []byte, name string) []mcfgv1.ControllerCertificate {
 			klog.Infof("Unable to decode cert %s into a pem block. Cert is either empty or invalid.", string(cert))
 			break
 		}
+		cert = next
 		c, err := x509.ParseCertificate(b.Bytes)
 		if err != nil {
 			klog.Infof("Malformed Cert, not syncing")
```

Comment:
Can you also add the reason why this wasn't synced?

I guess one concern I would have is we're technically skipping user certificates based on the x509 error silently here. Right now, even though the infinite loop is bad, at least the user "errors" in the sense that the sync can never complete. Now we would not sync in the cert. Maybe we should reflect the error up somehow?

Reply:
Agree with Jerry. Also for better troubleshooting, it will be good to add in the log cert that had issue.